openvpn/hier/usr/share/untangle/bin/openvpn-auth-user-pass

```diff
@@ -25,7 +26,16 @@ if (username == None):
 password = os.environ.get('password')
 if (password == None):
     raise Exception("Missing password")
-    
-result = openvpnApp.userAuthenticate(username,password)
+
+# Figure out the OTP password.
+# OpenVPN clients will combine password and OTP into the password string.
+# Format is "SCRV1:<passwd_b64>:<otp_b64>"
+otp = 0
+if password.startswith("SCRV1:"):
```

Comment:
maybe check for parts splitting as well as startswith, is the password encoded at this point (for non OTP users)?

Reply:
The password is encoded at this point, so this check works, even if a user has a password that starts with "SCRV1:"